zulipterminal/model.py

```diff
@@ -387,6 +387,29 @@ def get_messages(self, *,
         display_error_if_present(response, self.controller)
         return response['msg']
 
+    def _fetch_message_history(self, message_id: int) -> str:
+        """
+        Fetches message history for a message using its message_id.
+        """
+        response = self.client.get_message_history(message_id=message_id)
+        if response['result'] == 'success':
+            # NOTE: The instance attribute, message_history, only has the last
+            # accessed message edit history. Use the helper,
+            # get_message_history(), to fetch and access the message history at
+            # any point.
+            self.message_history = response['message_history']
+            return ''
+        display_error_if_present(response, self.controller)
+        return response['msg']
+
+    def get_message_history(self, message_id: int,
+                            ) -> List[Dict[str, Union[int, str]]]:
+        """
+        Returns message history for a message.
+        """
+        assert self._fetch_message_history(message_id=message_id) == ''
```

Comment:
Yes, it would. This should be fixed with `fetch_message_history()` returning an empty list now for failed request.

Reply:
This was to assert against any implementation loophole. However, as you mention, it would have failed the assert and update the footer when the server was down as well.

This should be fixed now with `fetch_message_history()` returning an empty list for failed request.

Reply:
This was to assert against any implementation loophole. However, as you mention, it would have failed the assert and update the footer when the server was down as well.

This should've been fixed now with `fetch_message_history()` returning an empty list for failed request.